src/components/App.js

```diff
@@ -1,45 +1,47 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 
-import { inputChange } from './../redux/actions';
-import { SearchInput } from './SearchInput';
-import { pressButton } from './../redux/actions';
+import { inputChange, setSearchKeyword } from './../redux/actions';
 
 import '../App.css';
 
 import Button from './Button';
+import { SearchInput } from './SearchInput';
+import { CategoriesList } from './CategoriesList';
 
 class App extends Component {
   render() {
-    const { pressButton, results, searchInput } = this.props;
+    const { 
+      searchInput, 
+      allCategories, 
+      selectedCategories,
+      setSearchKeyword,
+    } = this.props;
 
     return (
       <div className="App">
-        <div className="App-header">
-          <h2>Welcome to React</h2>
-        </div>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
         <SearchInput {...this.props} label="Search Events"/>
-      <Button 
-        className='hello'
-        aria-label='hello'
-        content='hello'
-        handleClick={ () => pressButton(searchInput) }
-      />
-      { results ? results.map(result => <li key={result.recipe.label}>{result.recipe.label}</li>) : null }
+        <Button 
+          className='hello'
+          aria-label='hello'
+          content='hello'
+          handleClick={ () => setSearchKeyword(searchInput) }
+        />
+        <CategoriesList allCategories={allCategories} selectedCategories={selectedCategories} />
       </div>
       
     );
   }
 }
 
 const connectConfig = connect(state => ({
-  searchInput: state.searchInput
+  searchInput: state.searchInput,
+  searchKeyword: state.categories.searchKeyword,
+  allCategories: state.categories.items,
+  selectedCategories: state.categories.items ? state.categories.items.filter(category => category.name.includes(state.categories.searchKeyword)) : null,
 }), {
```

Comment:
if we are storing the current category (searchKeyword) and allCategories in state, might it be better to do the filtering logic inside a component aka don't store selectedCategories- and keep the state as small/simple as possible? 

so in CategoriesList, you would just switch from this
`const categories = props.selectedCategories ? props.selectedCategories : props.allCategories; `

to this
`const categories = props.searchKeyword ? props.allCategories.filter(category => category.name.includes(props.searchKeyword)) : props.allCategories; `

and the rest as normal. Let me know what you think!